mesonbuild/compilers/mixins/gnu.py

```diff
@@ -219,6 +219,17 @@ def get_gui_app_args(self, value: bool) -> T.List[str]:
             return ['-mwindows' if value else '-mconsole']
         return []
 
+    def get_win_subsystem_args(self, value: str) -> T.List[str]:
+        if self.info.is_windows() or self.info.is_cygwin():
+            if 'windows' in value:
+                return ['-mwindows']
+            elif 'console' in value:
+                return ['-mconsole']
```

Comment:
Shouldn't there be a warning for something like `windows,6.0`, since the `6.0` part is ignored, or is this irrelevant for MinGW?

Reply:
GCC documentation does not say anything about specifying a version. So I don't really know. I'm guessing people who care about this detail will only use MSVC anyway.

Reply:
These things are controllable via `ld` flags: see `--subsystem`, `--major-subsystem-version`, `--minor-subsystem-version` in `man ld`. 

If you look at the output of `gcc -dumpspecs` you can see that `-mwindows` is a convenient shortcut for `--subsystem windows` and some libraries.

Reply:
Rewriting this but currently msys download page is unresponsive so I can't actually test it locally but instead have to use CI. :(